homeassistant/components/habitica/sensor.py

```diff
@@ -313,26 +250,27 @@ def name(self):
     @property
     def native_value(self):
         """Return the state of the device."""
-        return self._state
+        return sum(
+            1
+            for task in self.coordinator.data.tasks
+            if task.get("type") in self._task_type.path
+        )
```

Comment:
I think there must be a better way to count entries

Reply:
not sure if it is worth optimizing the counting algorithm for a list that has like 20 elements on average, maybe 50-100 elements at max. Also this will hopefully become deprecated with the todo entities

Reply:
I mean, doing a sum of a generator that spits out 1's looks really strange
```suggestion
        return len(list(
            for task in self.coordinator.data.tasks
            if task.get("type") in self._task_type.path
        ))
```
This would be more logical

Reply:
sure about this? now i get  `error: invalid syntax  [syntax]` from mypy